### PR TITLE
Unique anonymous ID statistics tracking

### LINKED ISSUE #16266: Addition of a unique anonymous installation ID with limited persistence
### Is your feature request related to a problem? Please describe.

In order to gain a better understanding of our user base, NV Access is going to add an optional, random, anonymous, rotating installation identifier to NVDA. This information is a powerful development and fundraising tool, as it helps us to identify usage patterns, allocate resources more effectively and is instrumental to our fundraising efforts.

### Describe the solution you'd like

If the user has opted-in to providing telemetry on update checks, the locally stored UUID is sent as part of the request. On the first of every month, all UUIDs are reset in order to maintain user anonymity and privacy.

### Describe alternatives you've considered

- A UUID stored in a local file
- A UUID stored in the windows registry
- A persistent vs. monthly rotating identifier
- A persistent hardware serial hash that is tied to hardware components

### Additional context

Changes will also be required in the NV Access analytics infrastructure to handle the additional information sent in the update check.

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Closes #16266 

### Summary of the issue:
Right now NV access is unable to gauge monthly or weekly users, only daily users.
Each day NVDA performs an update check and optionally sends more usage data information.
We are unable to differentiate copies performing update checks, meaning we only know X users who checked for an update on a given day.

### Description of user facing changes
None

### Description of development approach
A unique ID is stored to user config when initializing the update check.
This creates a unique ID per user of NVDA, counting portable copies and each Windows user's profile of an installation.

When checking for update, NVDA now sends the unique ID if usage tracking is enabled.

### Testing strategy:
 - [x] Tested running NVDA from installed copy, ensured unique ID persisted. Set `lastCheckUpdate` time to trigger new unique ID.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
  - Added a unique anonymous ID for improved NVDA usage statistics gathering.

- **Documentation**
  - Updated user guide to clarify data usage statistics dialog and user consent process.

- **Enhancements**
  - Improved update checking functionality with monthly unique ID updates.

- **Bug Fixes**
  - Refined NVDA behavior in the Python console for better performance.

### CODE REVIEW

File: source/updateCheck.py [hunk 1]
```diff
@@ -138,14 +140,16 @@ def checkForUpdate(auto: bool = False) -> Optional[Dict]:
 		# Available values of PROCESSOR_ARCHITEW6432 found in:
 		# https://docs.microsoft.com/en-gb/windows/win32/winprog64/wow64-implementation-details
 		"x64": os.environ.get("PROCESSOR_ARCHITEW6432") == "AMD64",
+		"ARM": os.environ.get("PROCESSOR_ARCHITEW6432") == "ARM64",
```
Comment (seanbudd) [inline source/updateCheck.py:143]: is this necessary given osArchitecture below?

**Review by gerald-hartig — APPROVED** [review]

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ Unicode CLDR has also been updated.
 * NVDA will now report figures with no accessible children, but with a label or description. (#14514)
 * When reading by line in browse mode, "caption" is no longer reported on each line of a long figure or table caption. (#14874)
 * In the Python console, the last unexecuted command will no longer be lost when moving in the input history. (#16653, @CyrilleB79)
+* A unique anonymous ID is now sent as part of NVDA usage statistics gathering. (#16266)
```
Comment (Qchristensen) [inline user_docs/en/changes.md:44]: Minor suggestion: Clarify that this is optional (could alternatively leave the start of the line the same and say "optional NVDA usage statistics gathering" at the end)

```suggestion
* An optional, unique, anonymous ID is now sent as part of NVDA usage statistics gathering. (#16266)

```

**Review by Qchristensen — APPROVED** [review]
Looks good.  Made one minor suggestion re the changes file, but otherwise fine.

File: user_docs/en/changes.md (hunk 2 above)
Comment (seanbudd) [reply user_docs/en/changes.md:44]:
```suggestion
* A unique anonymous ID is now sent as part of optional NVDA usage statistics gathering. (#16266)
```

**Comment by seanbudd** [comment]
@codeofdusk , yes it resets once a month on the first check of the month

**Pull request merged**